pkg/virt-controller/watch/migration.go

```diff
@@ -106,6 +110,12 @@ func NewMigrationController(templateService services.TemplateService,
 		UpdateFunc: c.updateMigration,
 	})
 
+	c.nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
Do you want to do this?  You might only want the `nodeInformer` for cache, not to trigger the `sync`.

Reply:
> You might only want the `nodeInformer` for cache, not to trigger the `sync`.

I'm not sure I understand what you mean.
If I delete the [run command to sync caches](https://github.com/kubevirt/kubevirt/blob/9a5075928111de2b840d31d9d8b8ee21930b5b25/pkg/virt-controller/watch/migration_test.go#L154) the test hangs forever and fails on timeout. Also if I don't refer to events regarding nodes (node creation / deletion etc') what will notify the the work-queue to get the relevant node into the informer's cache?

Do you maybe mean that there should only be an `addFunc` defined for `AddEventHandler()` (without `DeleteFunc`/`UpdateFunc`)? I can delete the `UpdateFunc`, but wouldn't it be problematic to remove `DeleteFunc` since we then provide a node from cache that has been deleted?

I'm new to all of this so probably missing something :)

Reply:
The informer store the objects to cache. You don't need to do anything else.
[See also ](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/shared_informer.go#L557)

`EventHandler` is for the case when you want to enqueue the object to the controller work-queue. Upon the work-queue the controller syncs the observed and desired state.

So I would propose to remove the registration of the handler.